test/core/telemetry/metrics_test.cc

```diff
@@ -669,6 +669,8 @@ TEST_F(MetricsTest, ParallelStatsPluginRegistrationAndLookup) {
       while (GlobalStatsPluginRegistry::GetStatsPluginsForChannel(
                  StatsPluginChannelScope("", "", endpoint_config_))
                  .size() < 10000) {
+        // Yield to avoid starving the register threads.
+        std::this_thread::yield();
```

Comment:
I'm wondering how did you find out the fix? It seems like the test flakiness depends on the OS' scheduling which seems like something we shouldn't be relying on. Did the test flakiness go away with this fix?

Reply:
Yes, it did. This was only happening on windows, and I verified this with a 1000x run. Reproduces many times without this, and no failure with this.

>  It seems like the test flakiness depends on the OS' scheduling which seems like something we shouldn't be relying on. 

I want to test concurrency so I can't add mutexes or condition variables, but I also don't want threads to starve. Telling the OS to schedule other threads was the way I thought of. I'm open to other approaches as well.